code/code/cmd/cmd_deathstroke.cc

```diff
@@ -5,222 +5,168 @@
-    percent = ((10 - (caster->getArmor() / 50)) << 1);
-    percent += victim->getDexReaction() * 5;
-    percent -= caster->getAgiReaction() * 5;
-    int bKnown2 = victim->getSkillValue(SKILL_DEATHSTROKE);
-    if (bKnown2 > 0 &&
-        victim->bSuccess(bKnown2 + percent, SKILL_DEATHSTROKE) &&
-         (i = victim->specialAttack(caster,SKILL_DEATHSTROKE)) &&
-         i != GUARANTEED_FAILURE) {
-      /* victim hits attacker vitals while attacker is exposed */
-      CF(SKILL_DEATHSTROKE);
-      if (victim->getPosition() > POSITION_STUNNED) {
-        act("$N exploits $n's vulnerable state with a quick hit to the heart.",
-                FALSE, caster, 0, victim, TO_NOTVICT);
-        act("While you are vulnerable, $N strikes you in the center of your torso.",
-                FALSE, caster, 0, victim, TO_CHAR);
-        act("You take advantage of $n's vulnerability for a cheap shot!", 
-               FALSE, caster, 0, victim, TO_VICT);
+    if (caster->reconcileDamage(victim, 0,sktype) == -1) 
+      return DELETE_VICT;
+  }
+  // Applying the debuff and, on success, the buff
+  if (wasSuccess) {
+    caster->affectJoin(caster, &aff1, AVG_DUR_YES, AVG_EFF_YES, FALSE);
+    caster->affectJoin(caster, &aff2, AVG_DUR_YES, AVG_EFF_YES, FALSE);
+  }
+  else {
+    caster->affectTo(&aff1, -1);
+  }
 
-        dam = 3*victim->GetMaxLevel();
-        dam += 3*victim->plotStat(STAT_CURRENT, STAT_STR, 0, 6, 3);
-        dam = ::number(1, dam);
-        dam = victim->getActualDamage(victim, NULL, dam, SKILL_DEATHSTROKE);
-        dam /= 2;
+  // Victim gets a chance at counter attack
+  int victimSkillValue = victim->getSkillValue(SKILL_DEATHSTROKE);
+  int victimSkillModifier = (victim->GetMaxLevel()-caster->GetMaxLevel())/2;
+  victimSkillModifier += victim->getDexReaction() * 5;
+  victimSkillModifier -= caster->getAgiReaction() * 5;
+  victimSkillModifier -= skillLevel/2;
 
-        TThing *ob = victim->heldInPrimHand();
-        if (ob && ob->isBluntWeapon())
-          sktype = DAMAGE_CAVED_SKULL;
-        if (victim->reconcileDamage(caster, dam,sktype) == -1)
-          return DELETE_THIS;
-      }
-    }
-  }
-  /* success make monster attack caster player as player is  now */
-  /* perceived as a greatest threat to the monster's livelyhood */
-  TBeing *cfight = caster->fight();
-  TBeing *vfight = victim->fight();
-  if (!wasSuccess){
-    // on failure, only set fighting if not already fighting
-     if (!cfight)
-       caster->setCharFighting(victim);
-     if (!vfight)
-       caster->setVictFighting(victim);
-  } else if (!vfight && caster->sameRoom(*victim)) {
-    // check sameRoom since may have fled from the damage
+  if (((victimSkillValue + victimSkillModifier)> 0) &&
+      victim->bSuccess(victimSkillValue + victimSkillModifier, SKILL_DEATHSTROKE) &&
+      victim->specialAttack(caster,SKILL_DEATHSTROKE) != GUARANTEED_FAILURE) {
+    // Successful counter attack
     if (victim->getPosition() > POSITION_STUNNED) {
```

Comment:
Kinda weird to check for skills first and stun last.. but I guess it doesn't make any difference

Reply:
Submitted an updated commit to re-order.